### PR TITLE
Update grpc, csi spec and few dependencies to latest

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

```release-note-none

```

### CODE REVIEW

**Comment by humblec** [comment]
Just to mention: `make` works on top of the PR.

File: go.mod [hunk 1]
```diff
@@ -3,13 +3,13 @@ module github.com/kubernetes-csi/external-provisioner
 go 1.12
```
Comment (msau42) [inline go.mod:3]: Does this need to be updated to 1.13?

Comment (humblec) [reply go.mod:3]: Yeah.. that need to be done..

**Comment by humblec** [comment]
@msau42 tested with go 1.13 and its passing.
 ```
                 WARNING

  This projects is tested with Go v1.13.
  Your current Go version is v1.14.
  This may or may not be close enough.

  In particular test-gofmt and test-vendor
  are known to be sensitive to the version of
  Go.
======================================================

mkdir -p bin
CGO_ENABLED=0 GOOS=linux go build  -a -ldflags '-X main.version=v1.6.0-11-ge1ba62f2c -extldflags "-static"' -o ./bin/csi-provisioner ./cmd/csi-provisioner
if [ "$ARCH" = "amd64" ]; then \
	CGO_ENABLED=0 GOOS=windows go build  -a -ldflags '-X main.version=v1.6.0-11-ge1ba62f2c -extldflags "-static"' -o ./bin/csi-provisioner.exe ./cmd/csi-provisioner ; \
	CGO_ENABLED=0 GOOS=linux GOARCH=ppc64le go build  -a -ldflags '-X main.version=v1.6.0-11-ge1ba62f2c -extldflags "-static"' -o ./bin/csi-provisioner-ppc64le ./cmd/csi-provisioner ; \
fi
```

**Comment by humblec** [comment]
@msau42 addressed the comment. PTAL.

File: go.mod [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-go 1.12
+go 1.13
 
 require (
-	github.com/container-storage-interface/spec v1.1.0
-	github.com/golang/mock v1.2.0
+	github.com/container-storage-interface/spec v1.2.0
+	github.com/golang/mock v1.4.3
 	github.com/kubernetes-csi/csi-lib-utils v0.7.0
 	github.com/kubernetes-csi/csi-test v2.0.0+incompatible
```
Comment (msau42) [inline go.mod:9]: Can we update csi-test and external-snapshotter to the latest released versions as well? I'm actually surprised snapshotter is still on 1.2.1? Didn't we need the new beta apis? cc @xing-yang

Comment (humblec) [reply go.mod]: @msau42 Updated csi test and also external snapshotter. As you noticed external-snpahotter was on 1.2.x :( .

**Comment by msau42** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**